### ISSUE TITLE
Starship `get_provider_info` required for newer Airflow Version

### ISSUE THREAD

**Issue by fritz-astronomer** [astronomer/starship#44]
FYI: Starship 1.0.6 does not work with Runtime 9.0.0
```
Webserver logs from my local CLI env:
 Traceback (most recent call last):
   File "/usr/local/bin/airflow", line 8, in <module>
     sys.exit(main())
              ^^^^^^
   File "/usr/local/lib/python3.11/site-packages/airflow/__main__.py", line 60, in main
     args.func(args)
   File "/usr/local/lib/python3.11/site-packages/airflow/cli/cli_config.py", line 49, in command
     return func(*args, **kwargs)
            ^^^^^^^^^^^^^^^^^^^^^
   File "/usr/local/lib/python3.11/site-packages/airflow/utils/cli.py", line 113, in wrapper
     return f(*args, **kwargs)
            ^^^^^^^^^^^^^^^^^^
   File "/usr/local/lib/python3.11/site-packages/airflow/utils/providers_configuration_loader.py", line 55, in wrapped_function
     ProvidersManager().initialize_providers_configuration()
   File "/usr/local/lib/python3.11/site-packages/airflow/providers_manager.py", line 367, in wrapped_function
     func(*args, **kwargs)
   File "/usr/local/lib/python3.11/site-packages/airflow/providers_manager.py", line 503, in initialize_providers_configuration
     self._initialize_providers_configuration()
   File "/usr/local/lib/python3.11/site-packages/airflow/providers_manager.py", line 516, in _initialize_providers_configuration
     self.initialize_providers_list()
   File "/usr/local/lib/python3.11/site-packages/airflow/providers_manager.py", line 367, in wrapped_function
     func(*args, **kwargs)
   File "/usr/local/lib/python3.11/site-packages/airflow/providers_manager.py", line 443, in initialize_providers_list
     self._discover_all_providers_from_packages()
   File "/usr/local/lib/python3.11/site-packages/airflow/providers_manager.py", line 551, in _discover_all_providers_from_packages
     provider_info = entry_point.load()()
                     ^^^^^^^^^^^^^^^^^^
   File "/usr/local/lib/python3.11/importlib/metadata/__init__.py", line 204, in load
     return functools.reduce(getattr, attrs, module)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 AttributeError: module 'astronomer.__init__' has no attribute 'get_provider_info'
 ```

**Comment by seanmuth** [comment]
FYI @fritz-astronomer this temporarily blocked a customer upgrade/deploy. Easy fix to just remove starship since they're already migrated, but still not great it fully blocks the migration since the `run-airflow-migrations` pod crashloops with this error.

**Comment by fritz-astronomer** [comment]
Another part of this should probably be to make starship not throw errors if there's an issue :( Although - this might be an Airflow bug? Partially?

**Comment by hkc-8010** [comment]
Rappi faced the same issue while upgrading one of the deployments to runtime version 9.1.0. However, removing the starship package resolves the issue. Below is the response from customer.

```
Great, thanks Henkumar, it works

Perhaps we are going to need "astronomer-starship" in the future to migrate connections and variables between environments.

Do you know if this module will be compatible in some moment?
```

ZD ticket: https://astronomer.zendesk.com/agent/tickets/30055

**Comment by hkc-8010** [comment]
Another customer facing the same issue with runtime version 9.1.0. 
ZD ticket for reference: https://astronomer.zendesk.com/agent/tickets/30377

**Comment by fritz-astronomer** [comment]
fixed in #54

**Comment by wolfier** [comment]
This issue is reported for Runtime 9.16.0.

```
alan@Alans-MacBook-Pro test % kubectl logs magnificent-quark-4867-scheduler-777477885f-nnqgn -c scheduler -p
Waiting for host: magnificent-quark-4867-pgbouncer.magnificent-quark-4867.svc.cluster.local 6543
Astro Runtime Version: 9.16.0
/usr/local/lib/python3.11/site-packages/airflow/plugins_manager.py:265 RemovedInAirflow3Warning: This decorator is deprecated.

In previous versions, all subclasses of BaseOperator must use apply_default decorator for the `default_args` feature to work properly.

In current version, it is optional. The decorator is applied automatically using the metaclass.

/usr/local/lib/python3.11/site-packages/airflow/plugins_manager.py:265 AirflowProviderDeprecationWarning: This module is deprecated. Please use `airflow.providers.cncf.kubernetes.operators.pod` instead.
Traceback (most recent call last):
  File "/usr/local/bin/airflow", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airflow/__main__.py", line 54, in main
    conf = write_default_airflow_configuration_if_needed()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airflow/configuration.py", line 1976, in write_default_airflow_configuration_if_needed
    conf.write(
  File "/usr/local/lib/python3.11/site-packages/airflow/configuration.py", line 677, in write
    with self.make_sure_configuration_loaded(with_providers=include_providers):
  File "/usr/local/lib/python3.11/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/airflow/configuration.py", line 531, in make_sure_configuration_loaded
    ProvidersManager()._initialize_providers_configuration()
  File "/usr/local/lib/python3.11/site-packages/airflow/providers_manager.py", line 516, in _initialize_providers_configuration
    self.initialize_providers_list()
  File "/usr/local/lib/python3.11/site-packages/airflow/providers_manager.py", line 367, in wrapped_function
    func(*args, **kwargs)
  File "/usr/local/lib/python3.11/site-packages/airflow/providers_manager.py", line 443, in initialize_providers_list
    self._discover_all_providers_from_packages()
  File "/usr/local/lib/python3.11/site-packages/airflow/providers_manager.py", line 551, in _discover_all_providers_from_packages
    provider_info = entry_point.load()()
                    ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/importlib_metadata/__init__.py", line 211, in load
    return functools.reduce(getattr, attrs, module)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'astronomer.__init__' has no attribute 'get_provider_info'
```
